safe_transaction_service/history/indexers/erc20_events_indexer.py

```diff
@@ -79,18 +81,25 @@ def _do_node_query(
         else:
             addresses_chunks = [addresses]
 
-        transfer_events = []
-        for addresses_chunk in addresses_chunks:
-            with self.auto_adjust_block_limit(from_block_number, to_block_number):
-                transfer_events.extend(
-                    self.ethereum_client.erc20.get_total_transfer_history(
-                        addresses_chunk,
-                        from_block=from_block_number,
-                        to_block=to_block_number,
-                    )
-                )
-
-        return transfer_events
+        jobs = []
+
+        gevent_pool = pool.Pool(self.get_logs_concurrency)
+        jobs = [
+            gevent_pool.spawn(
+                self.ethereum_client.erc20.get_total_transfer_history,
+                addresses_chunk,
+                from_block=from_block_number,
+                to_block=to_block_number,
+            )
+            for addresses_chunk in addresses_chunks
+        ]
+
+        with self.auto_adjust_block_limit(from_block_number, to_block_number):
+            # Check how long the first job takes
+            gevent.joinall(jobs[:1])
```

Comment:
AFAIK before we was adjusting the `block_limit` for every `addresses_chunk` but this new implementation adjusts only for the first job, could this affect to the performance? I mean, could the node change his performance and the first block_limit don't be the best choose for the next jobs?

Reply:
I assume the node for the in each batch first job would be chosen using round-robin, so maybe it's not accurate in the first execution but I think it will be converge to the average (if I'm not missing something).

Reply:
@moisses89 It's exactly what @hectorgomezv exposed. Maybe first time it's not very accurate, but at more requests are queued into the node it should stabilize